lucene/s3directory/src/test/org/apache/lucene/store/s3/TestS3Directory.java

```
@@ -0,0 +1,342 @@
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.index.StoredFields;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.store.FlushInfo;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.store.MMapDirectory;
+import org.apache.lucene.tests.util.LuceneTestCase;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestS3Directory extends LuceneTestCase {
+
+  private static final Logger logger = Logger.getLogger(TestS3Directory.class.getName());
+
+  public static final String TEST_BUCKET = "TEST-lucene-s3-directory-dir";
+  public static final String TEST_BUCKET1 = "TEST-lucene-s3-directory-dir1";
+  public static final String TEST_BUCKET2 = "TEST-lucene-s3-directory-dir2";
+
+  private S3Directory s3Directory;
+  private Directory fsDirectory;
+  private Directory ramDirectory;
+  private Analyzer analyzer = new SimpleAnalyzer();
+
+  private final Collection<String> docs = loadDocuments(3000, 5);
+  private final IndexWriterConfig.OpenMode openMode = IndexWriterConfig.OpenMode.CREATE_OR_APPEND;
+  private final boolean useCompoundFile = false;
+
+  @Override
+  public void setUp() {
+    s3Directory = new S3Directory(TEST_BUCKET, "", S3LockFactory.INSTANCE);
+    s3Directory.create();
+
+    try {
+      ramDirectory = new MMapDirectory(FileSystems.getDefault().getPath("target/index"));
+      fsDirectory = FSDirectory.open(FileSystems.getDefault().getPath("target/index"));
+    } catch (IOException ex) {
+      logger.log(Level.SEVERE, null, ex);
```

Let's throw this exception instead of swallowing it, we should fail the test suite if this throws an exception?